### PR TITLE
Add DNS client / connection

### PR DESCRIPTION
**Pull request by tseaver**
~~Uses #1137 as a base.~~

### CODE REVIEW

**Comment by dhermes** [comment]
Ping me when #1137 is done. Sorry for being AWOL, been traveling and will be "stable" by tomorrow afternoon.

**Comment by tseaver** [comment]
@dhermes rebased after merge of #1137.

File: gcloud/dns/connection.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    """The base of the API call URL."""
+
+    API_VERSION = 'v1'
+    """The version of the API, used in building the API call's URL."""
+
+    API_URL_TEMPLATE = '{api_base_url}/dns/{api_version}{path}'
+    """A template for the URL of a particular API call."""
+
+    SCOPE = ('https://www.googleapis.com/auth/ndev.clouddns.readwrite',
+             'https://www.googleapis.com/auth/cloud-platform')
```
Comment (dhermes) [inline gcloud/dns/connection.py]: Do we need both? `cloud-platform` gives the keys to the kingdom, so we should leave it out if we can

Comment (tseaver) [inline gcloud/dns/connection.py]: 23eadfc

**Comment by dhermes** [comment]
LGTM

**Comment by dhermes** [comment]
Looks like readthedocs.org has better support for simultaneous merges than Travis does: https://readthedocs.org/projects/gcloud-python/builds/3334189/

**Pull request merged**